src/Guzzle/MiddlewareFactory.php

```diff
@@ -53,7 +53,22 @@ public function retry( $delay = true ) {
 	 * @return callable
 	 */
 	private function getRetryDelay() {
-		return function( $numberOfRetries ) {
+		return function( $numberOfRetries, Response $response = null ) {
+			// The $response argument is only passed as of Guzzle 6.2.2.
+			if( $response !== null ) {
+				// Retry-After may be a number of seonds or an absolute date.
+				$retryAfter = $response->getHeaderLine( 'Retry-After' );
+
+				if( is_numeric( $retryAfter ) ) {
+					return 1000 * $retryAfter;
+				}
+
+				if( $retryAfter ) {
+					$seconds = strtotime( $retryAfter ) - time();
```

Comment:
The API doc doesn't mention in, but it is allowed according to [RFC 7231, section 7.1.3](https://tools.ietf.org/html/rfc7231#section-7.1.3), so I thought it was worth spending a few lines on.

The date returned by the API is a full date with timezone in a format supported by `strtotime()`, so timezones are not a problem. But it may cause issues if the local clock is very far behind – but I don't expect that to be a common problem.